app/backend/src/core/controllers/pricing/PriceUpdaterController.ts

```diff
@@ -1,127 +1,125 @@
-import { PriceRepository } from "@/database/repositories/PricingRepository";
-import { WhitelistedAsset } from "@/core/clients/coincap/assets/types";
-import whitelisted from "@/core/clients/coincap/assets/whitelisted.json";
-import CoinCapClient from "@/core/clients/coincap/CoinCapClient";
+import {
+  PriceRecord,
+  PriceRepository,
+} from "@/database/repositories/PricingRepository";
+import { WhitelistedAsset } from "@/core/clients/coingecko/assets/types";
+import whitelisted from "@/core/clients/coingecko/assets/whitelisted.json";
 import { UnixTime } from "@/core/types/UnixTime";
 import Logger from "@/tools/Logger";
+import { CoinGeckoClient } from "@/core/clients/coingecko/CoinGeckoClient";
+import { PricingModuleConfig } from "@/config/Config";
 
 export class PriceUpdaterController {
-  private readonly coinCapClient: CoinCapClient;
+  private readonly client: CoinGeckoClient;
   private readonly priceRepository: PriceRepository;
-  private readonly whitelistedAssets: WhitelistedAsset[];
   private readonly logger: Logger;
+  private readonly whitelistedAssets: WhitelistedAsset[];
+  private readonly backfillPeriodDays: number;
+  private readonly pollIntervalMs: number;
 
-  private priceIntervalMinutes: number;
-  private maxApiReturnDays: number;
-
-  private batchSize: number;
-  private batchDurationInMs: number;
+  private hasBackfilledHistory: boolean;
 
   constructor(
-    coinCapClient: CoinCapClient,
+    client: CoinGeckoClient,
     priceRepository: PriceRepository,
+    config: PricingModuleConfig,
     logger: Logger,
   ) {
-    this.coinCapClient = coinCapClient;
+    this.client = client;
     this.priceRepository = priceRepository;
     this.logger = logger;
-
     this.whitelistedAssets = whitelisted;
-    this.priceIntervalMinutes = 15;
-    this.maxApiReturnDays = 7;
-
-    this.batchSize =
-      (this.maxApiReturnDays * 24 * 60 * 60 * 1000) /
-      (this.priceIntervalMinutes * 60 * 1000);
-    this.batchDurationInMs =
-      this.batchSize * (this.priceIntervalMinutes * 60 * 1000);
+    this.backfillPeriodDays = config.backfillPeriodDays;
+    this.pollIntervalMs = config.pollIntervalMs;
+    this.hasBackfilledHistory = false;
   }
 
   async start() {
-    await this.updatePricesForAllAssets();
+    await this.updateActivePrices();
+    await this.backfillHistory();
   }
 
-  private async updatePricesForAllAssets() {
-    const latestBoundaries = await this.priceRepository.findLatestByToken(
-      UnixTime.now().add(-30, "days"),
-    );
+  private async backfillHistory() {
```

Comment:
Need to handle case of app being "paused" for some period, see https://github.com/l2beat/l2beat/blob/main/packages/backend/src/modules/tvl/PriceUpdater.ts#L90